src/pages/groupComparison/GroupComparisonMutationMapper.tsx

```diff
@@ -46,14 +50,127 @@ export default class GroupComparisonMutationMapper extends MutationMapper<
+                        <div style={{ fontWeight: 'bold', paddingBottom: 3 }}>
+                            (A){' '}
+                            {
+                                this.store.dataStore
+                                    .sortedFilteredGroupedData[0].group
+                            }
+                        </div>
+                        <DriverAnnotationProteinImpactTypeBadgeSelector
+                            filter={this.proteinImpactTypeFilter}
+                            counts={this.mutationCountsByProteinImpactTypeForGroup(
+                                0
+                            )}
+                            onSelect={this.onProteinImpactTypeSelect}
+                            onClickSettingMenu={this.props.onClickSettingMenu}
+                            annotatedProteinImpactTypeFilter={
+                                this.annotatedProteinImpactTypeFilter
+                            }
+                        />
+                        <hr style={{ marginTop: 10, marginBottom: 10 }}></hr>
+                        <div style={{ fontWeight: 'bold', paddingBottom: 3 }}>
+                            (B){' '}
+                            {
+                                this.store.dataStore
+                                    .sortedFilteredGroupedData[1].group
+                            }
+                        </div>
+                        <DriverAnnotationProteinImpactTypeBadgeSelector
+                            filter={this.proteinImpactTypeFilter}
+                            counts={this.mutationCountsByProteinImpactTypeForGroup(
+                                1
+                            )}
+                            onSelect={this.onProteinImpactTypeSelect}
+                            onClickSettingMenu={this.props.onClickSettingMenu}
+                            annotatedProteinImpactTypeFilter={
+                                this.annotatedProteinImpactTypeFilter
+                            }
+                        />
+                    </div>
+                ) : (
+                    <div
+                        style={{
+                            paddingBottom: 15,
+                            paddingTop: 15,
+                        }}
+                    >
+                        <div style={{ fontWeight: 'bold', paddingBottom: 3 }}>
+                            (A){' '}
+                            {
+                                this.store.dataStore
+                                    .sortedFilteredGroupedData[0].group
+                            }
+                        </div>
+                        <ProteinImpactTypeBadgeSelector
+                            filter={this.proteinImpactTypeFilter}
+                            counts={this.mutationCountsByProteinImpactTypeForGroup(
+                                0
+                            )}
+                            onSelect={this.onProteinImpactTypeSelect}
+                        />
```

Comment:
Similar to the comment above, this can be a function or component which takes group index and name as parameters

Reply:
Separated both to a new function proteinImpactTypeBadgeSelectorForGroup